### PR TITLE
xen: 4.8.3 -> 4.8.5, unbreak

### PR DESCRIPTION
**Pull request by danbst**
It was broken during GlusterFs upgrade: https://github.com/NixOS/nixpkgs/pull/66736

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @oxij @Shados

### CODE REVIEW

**Comment by flokli** [comment]
@danbst I got
```
hash mismatch in fixed-output derivation '/nix/store/ymn4pq2msn3k090g5zyy2g67s6p12q2c-qemu-xen':
  wanted: sha256:090ibcgs3xwmavk9yg2vaqr3xp9hirnfd3r40ccvrl49c5x58w3g
  got:    sha256:0lb7zd5nvr6znx47z93nbq4gj8xfb3622s8r2cvmpqmwnmlc3nd4
```

**Comment by flokli** [comment]
This might be due to various lines like 
```
fatal: invalid refspec '+refs/tags/v1.4.0^{}'
```

**Comment by danbst** [comment]
@flokli yes, ofborg failed as well. I'll look into that, maybe I've used `nix-prefetch-git` but should use TOFU instead

File: pkgs/applications/virtualization/xen/4.8.nix [hunk 1]
```diff
@@ -43,18 +43,23 @@ let
     sha256 = "0gaz93kb33qc0jx6iphvny0yrd17i8zhcl3a9ky5ylc2idz0wiwa";
   };
 
+  # Ported from
+  #"https://xenbits.xen.org/gitweb/?p=qemu-xen.git;a=patch;h=e014dbe74e0484188164c61ff6843f8a04a8cb9d";
+  #"https://xenbits.xen.org/gitweb/?p=qemu-xen.git;a=patch;h=0e3b891fefacc0e49f3c8ffa3a753b69eb7214d2";
+  qemuGlusterfs6Fix = ./qemu-gluster-6-compat.diff;
```
Comment (flokli) [inline pkgs/applications/virtualization/xen/4.8.nix]: nixpkgs policy generally is to not include petches if we can fetchpatch them. Can you change this?

Comment (danbst) [reply pkgs/applications/virtualization/xen/4.8.nix]: The patches don’t apply cleanly (xen 4.8.5 is from 2017 and patches are from 2019). Had to fix manually

Comment (flokli) [reply pkgs/applications/virtualization/xen/4.8.nix]: Fair enough then.

**Pull request merged**